draft-santesson-svt-jws.txt

```diff
@@ -0,0 +1,336 @@
+     4.2.  Signed Data Reference Data  . . . . . . . . . . . . . . .   4
+     4.3.  Signer Certificate References . . . . . . . . . . . . . .   4
+   5.  SVT JOSE Header . . . . . . . . . . . . . . . . . . . . . . .   5
+     5.1.  SVT Signing Key Reference . . . . . . . . . . . . . . . .   5
+   6.  IANA Considerations . . . . . . . . . . . . . . . . . . . . .   5
+     6.1.  Header Parameter Names Registration . . . . . . . . . . .   5
+       6.1.1.  Registry Contents . . . . . . . . . . . . . . . . . .   5
+   7.  Normative References  . . . . . . . . . . . . . . . . . . . .   5
+   Authors' Addresses  . . . . . . . . . . . . . . . . . . . . . . .   6
+
+1.  Introduction
+
+   The "Signature Validation Token" specification [SVT] defines the
+   basic token to support signature validation in a way that can
+   significantly extend the lifetime of a signature.
+
+   This specification defines a profile for implementing SVT with a JWS
+   signed payload according to [RFC7515], and defines the following
+   aspects of SVT usage:
+
+   *  How to include reference data related to JWS signatures in an SVT.
+
+   *  How to add an SVT token to JWS signatures.
+
+   A JWS may have one or more signatures depending on its serialization
+   format, signing the same payload data.  A JWS either contains the
+   data to be signed (enveloping) or may sign any externally associated
+   payload data (detached).
+
+   To provide a generic solution for JWS, an SVT is added to each
+   present signature as a JWS Unprotected Header.  If a JWS includes
+   multiple signatures, then each signature includes its own SVT.
+
+
+
+
+
+
+
+
+
+
+Santesson & Housley       Expires 6 March 2022                  [Page 2]
+
+Internet-Draft       JWS Signature Validation Token       September 2021
+
+
+2.  Definitions
+
+   The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT",
+   "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and
+   "OPTIONAL" in this document are to be interpreted as described in
+   BCP 14 [RFC2119] [RFC8174] when, and only when, they appear in all
+   capitals, as shown here.
+
+   The definitions in [SVT] and [RFC7515] apply also to this document.
+
+3.  SVT in JWS
+
+   An SVT token MAY be added to any signature of a JWS to support
+   validation of that signature.  If more than one signature is present
+   then each present SVT MUST provide information exclusively related to
+   one associated signature and MUST not include information about any
+   other signature in the JWS.
+
+   Each SVT is stored in its associated signature's "svt" header as
+   defined in Section 3.1.
+
+3.1.  "svt" Header Parameter
+
+   The "svt" (Signature Validation Token) Header Parameter is used to
+   contain an array of SVT tokens to support validation of the
+   associated signature.  Each SVT token in the array has the format of
+   a JWT as defined in [RFC7519] and is stored as a string object
```

Comment:
"and is stored as a string object without further wrapping or encoding"...

But a JWT is an JSON object, and you are talking about storing it as a string value. Wouldn't it be better to store it as an object, or if a string value is desired, use Base64-encoding?

Reply:
OK, I'll understand. The JWT **is** a string. Ignore the comment.